apis/server/image_bridge.go

```diff
@@ -32,8 +36,18 @@ func (s *Server) pullImage(ctx context.Context, rw http.ResponseWriter, req *htt
 		metrics.ImagePullSummary.WithLabelValues(image + ":" + tag).Observe(metrics.SinceInMicroseconds(start))
 	}(time.Now())
 
+	// get registry auth from Request header
+	authStr := req.Header.Get("X-Registry-Auth")
+	authConfig := types.AuthConfig{}
+	if authStr != "" {
+		data := base64.NewDecoder(base64.URLEncoding, strings.NewReader(authStr))
+		err := json.NewDecoder(data).Decode(&authConfig)
+		if err != nil {
+			authConfig = types.AuthConfig{}
```

Comment:
I do not think we can use the default empty one. I think it is better to return this error since it seems the input value has some issues. WDYT? @Ace-Tang

Reply:
yes, it make sense.